src/H5MM.c

```diff
@@ -347,14 +109,8 @@ H5MM_calloc(size_t size)
     /* Use FUNC_ENTER_NOAPI_NOINIT_NOERR here to avoid performance issues */
     FUNC_ENTER_NOAPI_NOINIT_NOERR
 
-    if (size) {
-#if defined H5_MEMORY_ALLOC_SANITY_CHECK
-        if (NULL != (ret_value = H5MM_malloc(size)))
-            HDmemset(ret_value, 0, size);
-#else  /* H5_MEMORY_ALLOC_SANITY_CHECK */
+    if (size)
```

Comment:
Same

Reply:
It's not necessary according to the standard. I didn't bother with looking carefully at existing behavior since I was just scissoring out the sanity checks, especially since the next step is to simply map H5MM calls to C standard library calls, but I can change this pretty easily.